File: sdk/typescript/rooch-sdk/src/utils/b64.test.ts
// Copyright (c) RoochNetwork
// SPDX-License-Identifier: Apache-2.0
import { describe, it, expect } from 'vitest'
import { toB64 } from './b64'

describe('toB64', () => {
  it('should convert small Uint8Array to base64 string correctly', () => {
    const input = new Uint8Array([72, 101, 108, 108, 111]) // "Hello"
    const expectedOutput = 'SGVsbG8='
    expect(toB64(input)).toBe(expectedOutput)
  })

  it('should return an empty string when input is an empty Uint8Array', () => {
    const input = new Uint8Array([])
    const expectedOutput = ''
    expect(toB64(input)).toBe(expectedOutput)
  })

  it('should process Uint8Array with length just below CHUNK_SIZE correctly', () => {
    const input = new Uint8Array(8191).fill(65) // "A" repeated 8191 times
    const expectedOutput = btoa(String.fromCharCode(...input))
    expect(toB64(input)).toBe(expectedOutput)
  })

  it('should handle Uint8Array with non-ASCII characters correctly', () => {
    const input = new Uint8Array([195, 164, 195, 182, 195, 188]) // "äöü" in UTF-8
    const expectedOutput = 'w6TDtsO8'
    expect(toB64(input)).toBe(expectedOutput)
  })
})

File: sdk/golang/rooch-sdk/utils/b64_test.go
package utils

import (
	"bytes"
	"testing"
)

func TestToB64(t *testing.T) {
	t.Run("should convert small byte slice to base64 string correctly", func(t *testing.T) {
		input := []byte("Hello")
		expectedOutput := "SGVsbG8="
		
		if result := ToB64(input); result != expectedOutput {
			t.Errorf("Expected %s but got %s", expectedOutput, result)
		}
	})

	t.Run("should return an empty string when input is empty", func(t *testing.T) {
		input := []byte{}
		expectedOutput := ""
		
		if result := ToB64(input); result != expectedOutput {
			t.Errorf("Expected empty string but got %s", result)
		}
	})

	t.Run("should process large byte slice correctly", func(t *testing.T) {
		input := bytes.Repeat([]byte("A"), 8191)
		result := ToB64(input)
		
		// Decode the result back to verify correctness
		decoded, err := base64.StdEncoding.DecodeString(result)
		if err != nil {
			t.Errorf("Failed to decode result: %v", err)
		}
		
		if !bytes.Equal(decoded, input) {
			t.Error("Decoded result doesn't match input")
		}
	})

	t.Run("should handle non-ASCII characters correctly", func(t *testing.T) {
		input := []byte{195, 164, 195, 182, 195, 188} // "äöü" in UTF-8
		expectedOutput := "w6TDtsO8"
		
		if result := ToB64(input); result != expectedOutput {
			t.Errorf("Expected %s but got %s", expectedOutput, result)
		}
	})
} 